pkg/match/path.go

```diff
@@ -291,31 +291,53 @@ func PathToScenes(ctx context.Context, name string, paths []string, sceneReader
 
 	filter.And = scene.PathsFilter(paths)
 
-	pp := models.PerPageAll
-	scenes, err := scene.Query(ctx, sceneReader, &filter, &models.FindFilterType{
-		PerPage: &pp,
-	})
+	// do in batches
+	pp := 1000
+	sort := "id"
+	sortDir := models.SortDirectionEnumAsc
+	lastID := 0
+
+	for {
+		if lastID != 0 {
+			filter.ID = &models.IntCriterionInput{
+				Value:    lastID,
+				Modifier: models.CriterionModifierGreaterThan,
+			}
+		}
 
-	if err != nil {
-		return nil, fmt.Errorf("error querying scenes with regex '%s': %s", regex, err.Error())
-	}
+		scenes, err := scene.Query(ctx, sceneReader, &filter, &models.FindFilterType{
+			PerPage:   &pp,
+			Sort:      &sort,
+			Direction: &sortDir,
+		})
 
-	var ret []*models.Scene
+		if err != nil {
+			return fmt.Errorf("error querying scenes with regex '%s': %s", regex, err.Error())
+		}
 
-	// paths may have unicode characters
-	const useUnicode = true
+		// paths may have unicode characters
+		const useUnicode = true
 
-	r := nameToRegexp(name, useUnicode)
-	for _, p := range scenes {
-		if regexpMatchesPath(r, p.Path) != -1 {
-			ret = append(ret, p)
+		r := nameToRegexp(name, useUnicode)
+		for _, p := range scenes {
+			if regexpMatchesPath(r, p.Path) != -1 {
+				if err := fn(ctx, p); err != nil {
+					return fmt.Errorf("processing scene %s: %w", p.GetTitle(), err)
```

Comment:
I am not sure i get this right. If an error happens during the processing of a single scene the tagging stops altogether due to the return?
Same goes for lines 383,443

Testing seems ok for tagging from a tag, studio, peformer.
Tested also with ~250k images and i can verify the original isssue #3091 is fixed

Reply:
That's correct. I don't see that as a huge issue. If we really don't want to stop processing then we can log the error in `fn` instead of returning the error from it.